Builders plan sidebar, intro content, onboarding workflows (#1300)
replicatedhq/replicated-docs: docs/vendor/replicated-onboarding.md

   1. Click **Promote**.

     :::note
     You can ignore any error messages in the **Promote Release** dialog.
     ::: 

1. Create a customer so that you can install the release in your cluster:

   1. Click **Customer > Create customer**.

   1. For **Customer name**, add a name.

   1. For **Channel**, select **Unstable**. This allows the customer to install releases promoted to the Unstable channel.

   1. For **Customer email**, enter an email address. An email address is required for Helm installations to authenticate with the Replicated registry.

   1. Click **Save Changes**.

   For more information, see [Creating and Managing Customers](/vendor/releases-creating-customer).

1. Install the application:
   1. On the **Customers** page for the customer that you created, click **Helm install instructions**.

      ![Helm install instructions button](/images/helm-install-button.png)

      [View a larger image](/images/helm-install-button.png)

   1. Run the commands in the **Helm install instructions** dialog to log in to the registry and install the Helm chart. Skip the step to run preflight checks.

>      <img alt="Helm install instructions dialog" src="/images/helm-install-instructions-no-preflights.png" width="500px"/>

Reviewer: broken

   
     This instance telemetry is automatically collected and sent back to the vendor portal when the Replicated SDK is installed alongside the application. For more information, see [Customer Reporting and Instance Insights](/vendor/replicated-sdk-overview#insights) in _About the Replicated SDK_.

   1. Under **Instance ID**, click on the ID to view additional insights including the versions of Kubernetes and the Replicated SDK running in the cluster where you installed the application. For more information, see [Instance Details](/vendor/instance-insights-details).

1. Create a new release in the vendor portal and then upgrade the instance in the cluster:

   1. Make a small change in the chart, such as incrementing the semantic version in the `Chart.yaml` to a new version. Then, package the chart again.

   1. In the vendor portal, create a new release (**Releases > Create release**). Drag and drop the new chart `.tgz` and then promote the new release to the Unstable channel. 

   1. In your cluster, run `helm upgrade` to upgrade the instance to the new release that you just promoted.

      **Example**:

      ```

>       helm upgrade --namespace default wordpress oci://registry.replicated.com/my-app/unstable/wordpress

Reviewer: `--namespace default` will not always be correct, and will depend what was passed in when the chart was initially installed. if no `namespace` was passed during install, then no `namespace` should be passed during upgrade

Further, I'd give a bit more guidance on how to construct the upgrade command, rather than just an example with no context

Author: Thanks, updated

      ```

      See [Helm Upgrade](https://helm.sh/docs/helm/helm_upgrade/) in the Helm documentation.

   1. After the upgrade completes, return to the **Instance details** page in the vendor portal and confirm that you can see the new application version.

      **Example**:

>       ![Instance details page](/images/onboarding-instance-details-new-version.png)

Reviewer: This screenshot should be updated once notifications is launched and the `gke-autopilot` bug is removed

1. (Recommended) Finish setting up your vendor portal account and team:

   1. If you are an admin, invite and manage team members. See [Invite Members](/vendor/team-management#invite-members) in _Managing Team Members_.

   1. Set up Slack or email notifications to be notified when there are changes in the installed instances of your application. Notifications can help catch problems before they happen and let you proactively contact customers to prevent support cases. See [Configuring Instance Notifications (Alpha)](/vendor/instance-notifications-config)

## Features Checklist

This section provides a checklist of key Replicated features to integrate with your application to fully onboard onto the Replicated platform. These features are provided in order of less challenging to more challenging, though you can configure and test the features in any order.

<table>
  <tr>
    <th width="20%">Feature</th>
    <th width="40%">Description</th>
    <th width="40%">How to</th>
  </tr>
  <tr>
    <td>Preflight checks</td>
    <td>
      <p>Define preflight checks to test for system compliance during the installation process and reduce the number of support escalations.</p>

>       <p><strong>Estimated time:</strong> 1 hour to define and test one or more collectors and analyzers for your application</p>

Reviewer: If this takes an hour, somethings not working well at all. However, I understand if you wanna overestimate these things. 

If you know what you're doing, this can be 10 minutes. 30m might be more accurate.

Also, in general, thoughts on linking to a lab on these topics as well? cc @crdant

Author: Talking to chuck tomorrow about linking to the labs https://replicated.slack.com/archives/C0522NKK988/p1690408824595079

    </td>
    <td>
      <ul>
      <li><a href="/vendor/preflight-helm-defining">Defining Preflight Checks for Helm Charts</a></li>
      <li><a href="/vendor/preflight-running">Running Preflight Checks for Helm Installations</a></li>
      </ul>
    </td>
  </tr>
  <tr>
    <td>Support bundles</td>
    <td>
      <p>Enable support bundles to collect and analyze troubleshooting data from your customers' clusters to help you diagnose problems with application deployments.</p>
      <p><strong>Estimated time:</strong> 1 hour to define and test one or more collectors and analyzers for your application</p>
    </td>
    <td>
      <ul>
      <li><a href="/vendor/support-bundle-helm-customizing">Customize Support Bundles for Helm Installations</a></li>
      <li><a href="/vendor/support-bundle-generating">Generating Support Bundles</a></li>
      </ul>
    </td>
  </tr>
  <tr>
    <td>Custom license entitlements</td>
    <td>

>       <p>Configure custom license fields fields that are specific to a customer, such as limiting the number of active users permitted.</p>

Reviewer: ```suggestion
      <p>Configure custom license fields that are specific to a customer, such as limiting the number of active users permitted.</p>
```

      <p><strong>Estimated time:</strong> 30 minutes to 1 hour to create and test each entitlement</p>
    </td>
    <td>
      <ul>
      <li><a href="/vendor/licenses-adding-custom-fields">Managing Custom License Fields</a></li>
      <li><a href="/vendor/licenses-reference-helm">Checking Entitlements for Helm Installations</a></li>
      </ul>
    </td>
  </tr>
  <tr>
    <td>Replicated SDK API</td>
    <td>

>       <p>Embed Replicated functionality into Helm chart applications.</p><p>To get started, use integration mode to develop locally without needing to make real changes in the vendor portal or in your environment.</p>

Reviewer: two things, you can take either, both, or neither

1. a tiny bit more detail on the "replicated functionality" that's covered here
2. tiny rephrase of "Helm Chart Application" -> "Helm based application"

```suggestion
      <p>Embed Replicated functionality like licensing and update checks into your Helm-based application.</p><p>To get started, use integration mode to develop locally without needing to make real changes in the vendor portal or in your environment.</p>
```

Author: I like em

      <p><strong>Estimated time:</strong> 1 hour to mock endpoints locally with integration mode, plus more time to optionally integrate features with your application</p>
    </td>
    <td>
      <ul>
      <li><a href="/reference/replicated-sdk-apis">Replicated SDK API</a></li>
      <li><a href="/vendor/replicated-sdk-development">Developing Against the SDK API</a></li>
      </ul>
    </td>
  </tr>
  <tr>
    <td>Proxy service</td>
    <td>
      <p>Allow customer licenses to grant proxy access to your application's private images.</p>
      <p><strong>Estimated time:</strong> 1 to 2 hours to connect your external registry and update your Helm chart to deliver image pull secrets for the proxy service</p>
    </td>
    <td>
      <a href="/vendor/helm-image-registry">Proxying Images for Helm Installations</a>
    </td>
  </tr>
  <tr>
    <td>Custom domains</td>

>     <td><p>Configure custom domains to alias the Replicated endpoints that are used for customer-facing URLs, such as the Replicated registry and proxy service.</p>

Reviewer: I think this would be more clear with actual endpoints

```suggestion
    <td><p>Configure custom domains to alias the Replicated endpoints that are used for customer-facing URLs, such as `registry.replicated.com` and `proxy.replicated.com`</p>
```

    <p><strong>Estimated time:</strong> 2 hours, including time to add entitlement checks in your application if you have not already</p></td>
    <td><a href="/vendor/licenses-verify-fields-sdk-api">Verifying License Field Signatures for Helm Installations</a>
    </td>
  </tr>
  <tr>
    <td>Compatibility matrix</td>
    <td><p>Quickly spin up ephemeral clusters for development, support, and testing.</p>
    <p><strong>Estimated time:</strong> Minutes to spin up a cluster with the replicated CLI, plus time for developing or testing on the cluster. Optionally, an additional 2 hours to configure GitHub Actions in your CI pipeline to automatically test releases with the compatibility matrix.</p>
    </td>
    <td>TBD
    </td>
  </tr>
  <tr>
    <td>Replicated KOTS</td>
    <td><p>For vendors with access to the KOTS installer, add custom resources to your release to support KOTS installations.</p>
    <p><strong>Estimated time:</strong> 1 to 2 hours to configure and test each custom resource.</p>
    </td>
    <td><a href="/vendor/distributing-workflow">Onboarding with KOTS</a>
    </td>
  </tr>
</table>

## Get Help from the Community

> The Replicated community site is a forum where Replicated team members and users can post questions and answers related to working with the Replicated platform. It is designed to help Replicated users troubleshoot and learn more about common tasks involved with distributing, installing, observing, and supporting their application. 

Reviewer: WDYT?

```suggestion
The [Replicated community site](https://community.replicated.com/) is a forum where Replicated team members and users can post questions and answers related to working with the Replicated platform. It is designed to help Replicated users troubleshoot and learn more about common tasks involved with distributing, installing, observing, and supporting their application. 
```

Author: Works for me!

## Workflow

This onboarding workflow provides steps for using the Replicated platform to create releases for a Helm chart-based application, then using the helm CLI to install in a development environment.

You will repeat these same basic steps to create and test releases throughout the onboarding process to test Replicated features. You will also repeat these steps after onboarding whenever you need to integrate new Replicated features with your application.

To begin onboarding to the Replicated platform with a Helm chart: 

1. Create an account in the vendor portal. You can either create a new team or join an existing team. For more information, see [Creating a Vendor Account](vendor-portal-creating-account).

1. Create a new application:

   1. In the [vendor portal](https://vendor.replicated.com/), from the application drop down, click **Create new app..**.

   1. In the **Name your application** field, enter a name based on the Helm chart that you will use for onboarding. For example, "Wordpress Test" or "MediaWiki Onboarding".

> 1. In the Helm chart `Chart.yaml`, add the Replicated SDK as a dependency:

Reviewer 2: We should add a note here that if the vendor's chart is using the Chart API version 1 (v1), then this would need to be added to the `requirements.yaml` file instead.  We can also point users to Helm's docs https://helm.sh/docs/faq/changes_since_helm2/#consolidation-of-requirementsyaml-into-chartyaml
